### PR TITLE
Move numerical test of Clip operator to OperatorTest

### PR DESCRIPTION
**Pull request by jygan**
*Description*:
 Don't check numerical results (done in OperatorTest)
 Verify expected node types for Clip

*Testing*: unit test passed
*Documentation*:
 no change

### CODE REVIEW

File: tests/unittests/OperatorTest.cpp [hunk 1]
```diff
@@ -2632,6 +2632,57 @@ TEST_P(InterpAndCPU, QuantizedTile) {
   }
 }
 
+TEST_P(InterpAndCPU, ClipWithDefaultValue) {
```
Comment (SplitInfinity) [inline tests/unittests/OperatorTest.cpp]: There's no need for this test, so you can delete it.

The default values are provided by the loader, not by the graph node itself, so this is not testing anything that the test below is not.

File: tests/unittests/OperatorTest.cpp [hunk 2]
```diff
@@ -2632,6 +2632,57 @@ TEST_P(InterpAndCPU, QuantizedTile) {
...
+  auto result = saveTensor->getHandle();
+  std::vector<size_t> expectedDims = {5, 5};
+  EXPECT_TRUE(result.dims().vec() == expectedDims);
+  for (size_t i = 0; i < 5 * 5; i++) {
+    EXPECT_FLOAT_EQ(result.raw(i), xHanle.raw(i));
+  }
+}
+
+TEST_P(InterpAndCPU, Clip) {
+  F_->setName("Clip");
```
Comment (SplitInfinity) [inline tests/unittests/OperatorTest.cpp]: There's no need for this; I suggest deleting it to follow the pattern set by the rest of the tests.

File: tests/unittests/OperatorTest.cpp [hunk 3]
```diff
@@ -2632,6 +2632,57 @@ TEST_P(InterpAndCPU, QuantizedTile) {
...
+  auto *X = mod_.createPlaceholder(ElemKind::FloatTy, {5, 5}, "X", false);
+  auto xHanle = ctx_.allocate(X)->getHandle();
```
Comment (SplitInfinity) [inline tests/unittests/OperatorTest.cpp]: `xHanle` -> `xHandle`

File: tests/unittests/Caffe2ImporterTest.cpp [hunk 4]
```diff
@@ -798,20 +794,20 @@ TEST(caffe2, importClip) {
...
-  for (size_t i = 0; i < 5 * 5; i++) {
-    EXPECT_FLOAT_EQ(result.raw(i), expectedValues[i]);
-  }
+  EXPECT_EQ(F->getNodes().size(), 5);
+  auto *saveNode = getSaveNodeFromDest(output);
+  auto *minNode = llvm::dyn_cast<MinNode>(saveNode->getInput().getNode());
+  ASSERT_TRUE(minNode);
+  auto *maxNode = llvm::dyn_cast<MaxNode>(minNode->getLHS().getNode());
+  ASSERT_TRUE(maxNode);
+  auto *maxSplatNode = llvm::dyn_cast<SplatNode>(minNode->getRHS().getNode());
```
Comment (SplitInfinity) [inline tests/unittests/Caffe2ImporterTest.cpp]: Can you also check that the input to this node is the same as the protobuf? You can access it using `maxSplatNode->getValue()`.

File: tests/unittests/Caffe2ImporterTest.cpp [hunk 5]
```diff
@@ -798,20 +794,20 @@ TEST(caffe2, importClip) {
...
+  ASSERT_TRUE(maxSplatNode);
+  auto *minSplatNode = llvm::dyn_cast<SplatNode>(maxNode->getRHS().getNode());
```
Comment (SplitInfinity) [inline tests/unittests/Caffe2ImporterTest.cpp]: Same here.

File: tests/unittests/Caffe2ImporterTest.cpp [hunk 6]
```diff
@@ -844,17 +836,20 @@ TEST(caffe2, importClipDefault) {
...
-  for (size_t i = 0; i < 5 * 5; i++) {
-    EXPECT_FLOAT_EQ(result.raw(i), inputs_0.getHandle().raw(i));
-  }
+  EXPECT_EQ(F->getNodes().size(), 5);
+  auto *saveNode = getSaveNodeFromDest(output);
+  auto *minNode = llvm::dyn_cast<MinNode>(saveNode->getInput().getNode());
+  ASSERT_TRUE(minNode);
+  auto *maxNode = llvm::dyn_cast<MaxNode>(minNode->getLHS().getNode());
+  ASSERT_TRUE(maxNode);
+  auto *maxSplatNode = llvm::dyn_cast<SplatNode>(minNode->getRHS().getNode());
```
Comment (SplitInfinity) [inline tests/unittests/Caffe2ImporterTest.cpp]: Can you also check that the input to this node is the expected default value? You can access it using `maxSplatNode->getValue()`.

File: tests/unittests/Caffe2ImporterTest.cpp [hunk 7]
```diff
@@ -844,17 +836,20 @@ TEST(caffe2, importClipDefault) {
...
+  ASSERT_TRUE(maxSplatNode);
+  auto *minSplatNode = llvm::dyn_cast<SplatNode>(maxNode->getRHS().getNode());
```
Comment (SplitInfinity) [inline tests/unittests/Caffe2ImporterTest.cpp]: Can you also check that the input to this node is the expected default value? You can access it using `maxSplatNode->getValue()`.

File: tests/unittests/OperatorTest.cpp (hunk 1 above)
Comment (jygan) [reply tests/unittests/OperatorTest.cpp]: yes, I agree, graph structure is the same.

File: tests/unittests/OperatorTest.cpp (hunk 3 above)
Comment (jygan) [reply tests/unittests/OperatorTest.cpp]: it's typo, good catch

**Comment by nadavrot** [comment]
Welcome to the project @jygan .

File: tests/unittests/Caffe2ImporterTest.cpp (hunk 4 above)
Comment (jygan) [reply tests/unittests/Caffe2ImporterTest.cpp]: @SplitInfinity  the protobuf means the input from  /glow/tests/models/caffe2Models/clip_op_net.pbtxt ?

**Comment by jygan** [comment]
@nadavrot thanks! this is my 1st open source project :-)

**Pull request merged**